sm_ipt/doc/usage.rst

```diff
@@ -0,0 +1,219 @@
+
+
+Initialization
+--------------
+
+The initialization function initializes context data and retrieves shared memory parameters from ``sm_ipt_os_init`` function.
+
+.. code-block:: c
+
+	struct sm_ipt_ctx sm_ipt_ctx;
+
+	void sm_ipt_receive_handler(const uint8_t *packet, size_t len)
+	{
+		/* Process data */
+		some_data_processing_function();
+	}
+
+	int initialize(void)
+	{
+		return sm_ipt_init(&sm_ipt_ctx, sm_ipt_receive_handler);
+	}
+
+The above code uses the ``sm_ipt_init`` function to initialize ``struct sm_ipt_ctx`` SM IPT context data structure and assign the receive handler.
+
+Data send full-copy
+-------------------
+
+The data is prepared beforehand, copied into allocated buffer and send to another core.
+
+.. code-block:: c
+
+	int send(const void *data, size_t len)
+	{
+		uint8_t *buffer;
+
+		sm_ipt_alloc_tx_buf(&sm_ipt_ctx, &buffer, len);
+
+		memcpy(buffer, data, len);
+
+		return sm_ipt_send(&sm_ipt_ctx, buffer, len);
+	}
+
+The above code uses the ``sm_ipt_alloc_tx_buf`` function to allocate transmit buffer in shared memory. Data is copied into said buffer and sent to another core via ``sm_ipt_send`` function.
+
+Data send no-copy
+-------------------
+
+The data is prepared after buffer allocation send to another core when data preparation (and usage) is finished.
+
+.. code-block:: c
+
+	int send_nocopy(size_t len)
+	{
+		uint8_t *buffer;
+		size_t i;
+
+		sm_ipt_alloc_tx_buf(&sm_ipt_ctx, &buffer, len);
+
+		/* Use buffer as any other allocated buffer */
+		some_data_preparation_function(buffer);
+
+		return sm_ipt_send(&sm_ipt_ctx, buffer, len);
+	}
+
+The above code uses the ``sm_ipt_alloc_tx_buf`` function to allocate transmit buffer in shared memory. Buffer is prepared by ``some_data_preparation_function`` and sent to another core via ``sm_ipt_send`` function.
+
+Data send abort
+-------------------
+
+There is possibility to free transmit buffer instead of sending it if the need arises.
+
+.. code-block:: c
+
+	int send_if(size_t len, bool send_choice)
+	{
+		uint8_t *buffer;
+		size_t i;
+
+		sm_ipt_alloc_tx_buf(&sm_ipt_ctx, &buffer, len);
+
+		if (send_choice){
+			/* Prepare some data */
+			some_data_preparation_function(buffer);
+
+			return sm_ipt_send(&sm_ipt_ctx, buffer, len);
+		}
+		else{
+			sm_ipt_free_tx_buf(&sm_ipt_ctx, buffer);
+
+			return ERROR_ABORT;
+		}
+	}
+
+The above code uses the ``sm_ipt_alloc_tx_buf`` function to allocate transmit buffer in shared memory. Buffer is send only if send_choice is true. If send_choice is false the buffer is free using ``sm_ipt_free_tx_buf`` and send operation is aborted.
+
+
+Data receive full-copy
+-------------------
+
+When data is received it can be processed using full-copy mechanism. Buffer has to be copied to local memory. Shared memory rx buffer has to be freed and data processing is done on local copy of buffer.
```

Comment:
```suggestion
When data is received, it can be processed using the full-copy mechanism.
The buffer must be copied to the local memory.
The shared memory rx buffer must be freed and data processing takes place on the local copy of the buffer.
```